services/verifier-alliance/main.mjs

```diff
@@ -0,0 +1,240 @@
+import pg from "pg";
+import fs from "fs";
+import path from "path";
+import { fileURLToPath } from "url";
+import dotenv from "dotenv";
+
+dotenv.config();
+
+const __filename = fileURLToPath(import.meta.url); // get the resolved path to the file
+const __dirname = path.dirname(__filename);
+
+const { Client } = pg;
+
+// Load current verified contract counter from file
+const COUNTER_FILE = path.join(__dirname, "CURRENT_VERIFIED_CONTRACT");
+let CURRENT_VERIFIED_CONTRACT = 1;
+if (fs.existsSync(COUNTER_FILE)) {
+  CURRENT_VERIFIED_CONTRACT = parseInt(
+    fs.readFileSync(COUNTER_FILE, "utf8"),
+    10
+  );
+}
+
+const N = 200; // Number of contracts to process at a time
+
+const SOURCE_DB_CONFIG = {
+  host: process.env.SOURCIFY_HOST,
+  database: process.env.SOURCIFY_DB,
+  user: process.env.SOURCIFY_USER,
+  password: process.env.SOURCIFY_PASSWORD,
+  port: process.env.SOURCIFY_PORT,
+};
+
+const TARGET_DB_CONFIG = {
+  host: process.env.VERA_HOST,
+  database: process.env.VERA_DB,
+  user: process.env.VERA_USER,
+  password: process.env.VERA_PASSWORD,
+  port: process.env.VERA_PORT,
+};
+
+const sourceClient = new Client(SOURCE_DB_CONFIG);
+const targetClient = new Client(TARGET_DB_CONFIG);
+
+async function upsertAndGetId(
+  query,
+  conflictQuery,
+  insertValues,
+  selectValues,
+  client
+) {
+  await client.query(query, insertValues);
+  const { rows } = await client.query(conflictQuery, selectValues);
+  return rows[0].id;
+}
+
+(async () => {
+  try {
+    await sourceClient.connect();
+    await targetClient.connect();
+
+    const { rows: verifiedContracts } = await sourceClient.query(
+      `
+            SELECT * FROM verified_contracts
+            WHERE id >= $1
+            ORDER BY id ASC
+            LIMIT $2
+        `,
+      [CURRENT_VERIFIED_CONTRACT, N]
+    );
+
+    for (const contract of verifiedContracts) {
+      const {
+        id: verifiedContractId,
+        deployment_id,
+        compilation_id,
+      } = contract;
+
+      // Update the CURRENT_VERIFIED_CONTRACT counter
+      CURRENT_VERIFIED_CONTRACT = verifiedContractId;
+      fs.writeFileSync(
```

Comment:
Wouldn't it be safer to only update this counter once we are sure the contract was written to the VerA database? If something goes wrong later, we might skip a contract, no?

Reply:
the code is 

```
            WHERE id >= $1
            ORDER BY id ASC
            LIMIT $2
        `,
      [CURRENT_VERIFIED_CONTRACT, N]
```

so it starts from the last CURRENT_VERIFIED_CONTRACT. If the scripts breaks while synching a verified_contract then it starts again from that contract.

Reply:
Ah, I misunderstood it. Looks fine then :)